engine/src/cdata.cpp

```diff
@@ -191,10 +191,16 @@ IO_stat MCCdata::save(IO_handle stream, Object_type type, uint4 p_part, uint32_t
 		}
 		else
 		{
+            // Ensure field's saved MCCdata paragraphs have a parent
+            if (p_parent != nil)
+                MCAssert(p_parent -> gettype() == CT_FIELD);
+            
 			MCParagraph *tptr = (MCParagraph *)data;
 			if (tptr != NULL)
 				do
 				{
+                    if (p_parent != nil)
+                        tptr -> setparent(static_cast<MCField *>(p_parent));
```

Comment:
If you use

```c++
tptr->setparent(MCObjectCast<MCField>(p_parent));
```

it will include the assertion that the parent is actually a field.